src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/GitLabService.java

```diff
@@ -101,17 +101,24 @@ public void addMemberToRepository(VcsRepositoryUrl repositoryUrl, User user) {
         final var userId = gitLabUserManagementService.getUserId(user.getLogin());
 
         try {
+            if (userAlreadyMember(repositoryId, userId)) return;
             gitlab.getProjectApi().addMember(repositoryId, userId, DEVELOPER);
+        } catch (GitLabApiException e) {
+            throw new GitLabException("Error while trying to add user to repository: " + user.getLogin() + " to repo " + repositoryUrl, e);
         }
-        catch (GitLabApiException e) {
-            if (e.getValidationErrors().containsKey("access_level")
-                    && e.getValidationErrors().get("access_level").stream().anyMatch(s -> s.contains("should be greater than or equal to"))) {
```

Comment:
check does not work is outdated

Reply:
Turns out that is not true. But it only happens if you try to lower the access level.

Reply:
so this needs to stay as well